src/features/activity/ActivityDetailsScreen.tsx

```diff
@@ -0,0 +1,297 @@
+    const firstNativeTransfer = enrichedTx.nativeTransfers[0]
+    const fromAccount =
+      firstTokenTransfer?.fromUserAccount ||
+      firstNativeTransfer?.fromUserAccount
+    const toAccount =
+      firstTokenTransfer?.toUserAccount || firstNativeTransfer?.toUserAccount
+
+    return (
+      <Box marginTop="l" flex={1} width="100%">
+        {fromAccount && (
+          <AddressActivityItem
+            accountAddress={fromAccount}
+            marginHorizontal="l"
+            marginBottom="xs"
+            borderRadius="xl"
+            showBubbleArrow
+            onMenuPress={onAddressItemPress(fromAccount)}
+          />
+        )}
+        {toAccount && (
+          <AddressActivityItem
+            accountAddress={toAccount}
+            marginHorizontal="l"
+            borderRadius="xl"
+            onMenuPress={onAddressItemPress(toAccount)}
+          />
+        )}
+      </Box>
+    )
+  }, [onAddressItemPress, transaction])
+
+  const title = useMemo(() => {
+    const enrichedTx = transaction as EnrichedTransaction
+    const confirmedSig = transaction as ConfirmedSignatureInfo
+
+    if (confirmedSig.err) {
+      return t('activityScreen.transactionFailed')
+    }
+
+    const txnKey = `activityScreen.enrichedTransactionTypes.${enrichedTx.type}`
+    return i18n.exists(txnKey)
+      ? t(`activityScreen.enrichedTransactionTypes.${enrichedTx.type}`)
+      : t('activityScreen.enrichedTransactionTypes.UNKNOWN')
+  }, [i18n, t, transaction])
+
+  const description = useMemo(() => {
+    const enrichedTx = transaction as EnrichedTransaction
+    const confirmedSig = transaction as ConfirmedSignatureInfo
+
+    // Custom description that elipiizes the address
+    if (enrichedTx.description) {
+      const customDescription = enrichedTx.description
+        ?.split(' ')
+        .map((word) => {
+          // Remove addresses
+          if (solAddressIsValid(word)) {
+            return ''
+          }
+          return word
+        })
+        .join(' ')
+        .trim()
+
+      // capitalize first letter of description
+      return `${customDescription
+        .charAt(0)
+        .toUpperCase()}${customDescription.slice(1)}`
+    }
+
+    if (confirmedSig.err) {
+      return 'Error'
+    }
+
+    return 'Transaction Successful'
+  }, [transaction])
+
+  const handleOpenExplorer = useCallback(async () => {
+    const url = `https://explorer.solana.com/tx/${transaction.signature}?cluster=${cluster}`
+    // Open url in browser
+    await Linking.openURL(url)
+  }, [cluster, transaction.signature])
+
+  const toggleActionSheet = useCallback(
+    (open) => () => {
+      setOptionsOpen(open)
+    },
+    [],
+  )
+
+  const showToast = useCallback(() => {
+    if (!selectedAddress) return
+    Toast.show(
+      t('generic.copied', {
+        target: ellipsizeAddress(selectedAddress),
+      }),
+    )
+  }, [selectedAddress, t])
+
+  const handleCopyAddress = useCallback(() => {
```

Comment:
You should be able to use the `useCopyText` hook here
```ts
const copyText = useCopyText()`

  copyText({
                message: ellipsizeAddress(selectedAddress),
                copyText: selectedAddress,
              })

Reply:
So many useful hooks 😅

Reply:
Done